diff --git a/src/dftbp/extlibs/magma.F90 b/src/dftbp/extlibs/magma.F90
index 59549f16..15bde3ab 100644
--- a/src/dftbp/extlibs/magma.F90
+++ b/src/dftbp/extlibs/magma.F90
@@ -11,7 +11,8 @@
 module dftbp_extlibs_magma
   use, intrinsic :: iso_c_binding, only : c_int
 #:if WITH_MAGMA
-  use magma, only : magmaf_ssygvd_m, magmaf_dsygvd_m, magmaf_chegvd_m, magmaf_zhegvd_m
+  use magma, only : magmaf_ssygvd_m, magmaf_dsygvd_m, magmaf_chegvd_m, magmaf_zhegvd_m,&
+      & magmaf_ssygvd, magmaf_dsygvd, magmaf_chegvd, magmaf_zhegvd
 #:endif
   implicit none
 
@@ -20,6 +21,7 @@ module dftbp_extlibs_magma
 #:if WITH_MAGMA
   public :: getGpusAvailable, getGpusRequested
   public :: magmaf_ssygvd_m, magmaf_dsygvd_m, magmaf_chegvd_m, magmaf_zhegvd_m
+  public :: magmaf_ssygvd, magmaf_dsygvd, magmaf_chegvd, magmaf_zhegvd
 #:endif
 
   !> Whether code was built with GPU support
diff --git a/src/dftbp/math/eigensolver.F90 b/src/dftbp/math/eigensolver.F90
index 2b70e6e1..11ada603 100644
--- a/src/dftbp/math/eigensolver.F90
+++ b/src/dftbp/math/eigensolver.F90
@@ -17,7 +17,7 @@ module dftbp_math_eigensolver
   use dftbp_io_message, only : error, warning
 #:if WITH_MAGMA
   use dftbp_extlibs_magma,  only : magmaf_ssygvd_m, magmaf_dsygvd_m, magmaf_chegvd_m,&
-      & magmaf_zhegvd_m
+      & magmaf_zhegvd_m, magmaf_ssygvd, magmaf_dsygvd, magmaf_chegvd, magmaf_zhegvd
 #:endif
   implicit none
 
@@ -2085,11 +2085,19 @@ contains
     allocate(rwork(1))
   #:endif
 
-    call magmaf_${NAME}$_m(ngpus, iitype, jobz, uplo, n, a, n, b, n, w, work, -1,&
-      #:if VTYPE == 'complex'
-        & rwork, -1,&
-      #:endif
-        & iwork, -1, info)
+    if (ngpus == 1) then
+      call magmaf_${NAME}$(iitype, jobz, uplo, n, a, n, b, n, w, work, -1,&
+        #:if VTYPE == 'complex'
+          & rwork, -1,&
+        #:endif
+          & iwork, -1, info)
+    else
+      call magmaf_${NAME}$_m(ngpus, iitype, jobz, uplo, n, a, n, b, n, w, work, -1,&
+        #:if VTYPE == 'complex'
+          & rwork, -1,&
+        #:endif
+          & iwork, -1, info)
+    end if
 
     if (info /= 0) then
       call error("Failure in MAGMA_${NAME}$ to determine optimum workspace")
@@ -2111,24 +2119,32 @@ contains
    #:endif
 
     ! MAGMA Diagonalization
-    call magmaf_${NAME}$_m(ngpus, iitype, jobz, uplo, n, a, n, b, n, w, work, lwork,&
-      #:if VTYPE == 'complex'
-        & rwork, lrwork,&
-      #:endif
-        & iwork, liwork, info)
+    if (ngpus == 1) then
+      call magmaf_${NAME}$(iitype, jobz, uplo, n, a, n, b, n, w, work, lwork,&
+        #:if VTYPE == 'complex'
+          & rwork, lrwork,&
+        #:endif
+          & iwork, liwork, info)
+    else
+      call magmaf_${NAME}$_m(ngpus, iitype, jobz, uplo, n, a, n, b, n, w, work, lwork,&
+        #:if VTYPE == 'complex'
+          & rwork, lrwork,&
+        #:endif
+          & iwork, liwork, info)
+    end if
 
     ! test for errors
     if (info /= 0) then
       if (info < 0) then
-        write(error_string, "('Failure in diagonalisation routine magmaf_${NAME}$_m,&
+        write(error_string, "('Failure in diagonalisation routine magmaf_${NAME}$,&
             & illegal argument at position ',i6)") info
         call error(error_string)
       else if (info <= n) then
-        write(error_string, "('Failure in diagonalisation routine magmaf_${NAME}$_m,&
+        write(error_string, "('Failure in diagonalisation routine magmaf_${NAME}$,&
             & diagonal element ',i6,' did not converge to zero.')") info
         call error(error_string)
       else
-        write(error_string, "('Failure in diagonalisation routine magmaf_${NAME}$_m,&
+        write(error_string, "('Failure in diagonalisation routine magmaf_${NAME}$,&
             & non-positive definite overlap! ',i6)") info - n
         call error(error_string)
       endif
