diff --git a/src/dftbp/common/environment.F90 b/src/dftbp/common/environment.F90
index 096d5ddc..97365b4f 100644
--- a/src/dftbp/common/environment.F90
+++ b/src/dftbp/common/environment.F90
@@ -47,13 +47,10 @@ module dftbp_common_environment
     type(TTimerArray), public, allocatable :: globalTimer
 
   #:if WITH_MPI
-
-    !> Global mpi settings
     type(TMpiEnv), public :: mpi
-
-    !> Whether MPI environment had been initialised
+    type(TMpiEnv), public :: mpiPostScc
     logical :: mpiInitialised = .false.
-
+    logical, public :: mpiPostSccInitialised = .false.
   #:endif
 
   #:if WITH_SCALAPACK
@@ -80,6 +77,7 @@ module dftbp_common_environment
 
   #:if WITH_MPI
     procedure :: initMpi => TEnvironment_initMpi
+    procedure :: initMpiPostScc => TEnvironment_initMpiPostScc
   #:endif
 
   #:if WITH_SCALAPACK
@@ -183,6 +181,10 @@ contains
     #:endif
 
     #:if WITH_MPI
+      if (this%mpiPostSccInitialised) then
+        call TMpiEnv_final(this%mpiPostScc)
+        this%mpiPostSccInitialised = .false.
+      end if
       if (this%mpiInitialised) then
         call TMpiEnv_final(this%mpi)
         this%mpiInitialised = .false.
@@ -252,6 +254,14 @@ contains
 
   end subroutine TEnvironment_initMpi
 
+
+  subroutine TEnvironment_initMpiPostScc(this, nGroup)
+    class(TEnvironment), intent(inout) :: this
+    integer, intent(in) :: nGroup
+    call TMpiEnv_init(this%mpiPostScc, globalMpiComm=this%mpi%globalComm, nGroup=nGroup)
+    this%mpiPostSccInitialised = .true.
+  end subroutine TEnvironment_initMpiPostScc
+
 #:endif
 
 
diff --git a/src/dftbp/dftbplus/initprogram.F90 b/src/dftbp/dftbplus/initprogram.F90
index 5c662bd2..425f986c 100644
--- a/src/dftbp/dftbplus/initprogram.F90
+++ b/src/dftbp/dftbplus/initprogram.F90
@@ -1453,6 +1453,8 @@ contains
     end if
 
     call env%initMpi(input%ctrl%parallelOpts%nGroup)
+    call env%initMpiPostScc(merge(input%ctrl%parallelOpts%nGroupPostScc,&
+        & input%ctrl%parallelOpts%nGroup, input%ctrl%parallelOpts%nGroupPostScc /= 0))
   #:endif
 
 
diff --git a/src/dftbp/dftbplus/inputdata.F90 b/src/dftbp/dftbplus/inputdata.F90
index 019e1d81..961600b4 100644
--- a/src/dftbp/dftbplus/inputdata.F90
+++ b/src/dftbp/dftbplus/inputdata.F90
@@ -70,6 +70,7 @@ module dftbp_dftbplus_inputdata
 
     !> Number of processor groups
     integer :: nGroup
+    integer :: nGroupPostScc = 0
 
     !> Blacs options
     type(TBlacsOpts) :: blacsOpts
diff --git a/src/dftbp/dftbplus/main.F90 b/src/dftbp/dftbplus/main.F90
index bf972bb0..2d3cfc27 100644
--- a/src/dftbp/dftbplus/main.F90
+++ b/src/dftbp/dftbplus/main.F90
@@ -15,6 +15,9 @@ module dftbp_dftbplus_main
   use dftbp_common_environment, only : TEnvironment, globalTimers
   use dftbp_common_file, only : TFileDescr, openFile, closeFile
   use dftbp_common_globalenv, only : stdOut, withMpi
+#:if WITH_MPI
+  use dftbp_common_mpienv, only : TMpiEnv
+#:endif
   use dftbp_common_hamiltoniantypes, only : hamiltonianTypes
   use dftbp_common_status, only : TStatus
   use dftbp_derivs_numderivs2, only : TNumderivs, next, getHessianMatrix, dipoleAdd, polAdd
@@ -1010,6 +1013,9 @@ contains
     logical :: constrConverged
 
     integer :: iSpin, iKS, iConstrIter, nConstrIter
+#:if WITH_MPI
+    type(TMpiEnv) :: savedMpi
+#:endif
 
     if (this%tDipole) allocate(dipoleTmp(3))
 
@@ -1472,6 +1478,12 @@ contains
       end block
     end if
 
+  #:if WITH_MPI
+    if (env%mpiPostSccInitialised) then
+      savedMpi = env%mpi ; env%mpi = env%mpiPostScc
+    end if
+  #:endif
+
     call env%globalTimer%startTimer(globalTimers%postSCC)
 
     if (this%isLinResp) then
@@ -1619,6 +1631,10 @@ contains
       call writeEsp(this%electrostatPot, env, iGeoStep, this%nGeoSteps)
     end if
 
+  #:if WITH_MPI
+    if (env%mpiPostSccInitialised) env%mpi = savedMpi
+  #:endif
+
   end subroutine processGeometry
 
 
diff --git a/src/dftbp/dftbplus/parser.F90 b/src/dftbp/dftbplus/parser.F90
index 919a6cf3..db6fed91 100644
--- a/src/dftbp/dftbplus/parser.F90
+++ b/src/dftbp/dftbplus/parser.F90
@@ -7543,6 +7543,7 @@ contains
       end if
       allocate(input%ctrl%parallelOpts)
       call getChildValue(node, "Groups", input%ctrl%parallelOpts%nGroup, 1, child=pTmp)
+      call getChildValue(node, "GroupsPostScc", input%ctrl%parallelOpts%nGroupPostScc, 0)
       call getChildValue(node, "UseOmpThreads", input%ctrl%parallelOpts%tOmpThreads, .not. withMpi)
       call readBlacs(node, input%ctrl%parallelOpts%blacsOpts)
     end if
