diff --git a/src/dftbp/common/globalenv.F90 b/src/dftbp/common/globalenv.F90
index d2184d6a..8dd7f9a5 100644
--- a/src/dftbp/common/globalenv.F90
+++ b/src/dftbp/common/globalenv.F90
@@ -1,6 +1,6 @@
 !--------------------------------------------------------------------------------------------------!
 !  DFTB+: general package for performing fast atomistic simulations                                !
-!  Copyright (C) 2006 - 2023  DFTB+ developers group                                               !
+!  Copyright (C) 2006 - 2025  DFTB+ developers group                                               !
 !                                                                                                  !
 !  See the LICENSE file for terms of usage and distribution.                                       !
 !--------------------------------------------------------------------------------------------------!
@@ -15,11 +15,11 @@
 !> code.
 !>
 module dftbp_common_globalenv
-  use, intrinsic :: iso_fortran_env, only : output_unit, error_unit
+  use, intrinsic :: iso_fortran_env, only : error_unit, output_unit
 #:if WITH_MPI
   use mpi, only : MPI_COMM_WORLD
-  use dftbp_extlibs_mpifx, only : mpifx_comm, MPI_THREAD_FUNNELED, mpifx_init_thread, mpifx_abort,&
-      & mpifx_barrier, mpifx_finalize
+  use dftbp_extlibs_mpifx, only : MPI_THREAD_FUNNELED, MPI_THREAD_MULTIPLE,&
+      & mpifx_abort, mpifx_barrier, mpifx_comm, mpifx_finalize, mpifx_init_thread
 #:endif
   implicit none
 
@@ -115,7 +115,7 @@ contains
       doMpiFinalization = .false.
     else
       mpiComm0 = MPI_COMM_WORLD
-      call mpifx_init_thread(requiredThreading=MPI_THREAD_FUNNELED)
+      call mpifx_init_thread(requiredThreading=MPI_THREAD_MULTIPLE)
     end if
 
     call globalMpiComm%init(commid=mpiComm0)
